x-pack/plugins/infra/public/containers/logs/log_source/log_source.ts

```diff
@@ -45,6 +47,10 @@ export const useLogSource = ({
         return await callFetchLogSourceConfigurationAPI(sourceId, fetch);
       },
       onResolve: ({ data }) => {
+        if (!getIsMounted()) {
```

Comment:
I assume this fixes the race condition you mentioned in the description.

Could you help me understand what was the problem? I see that [`useMountedState`](https://github.com/streamich/react-use/blob/master/src/useMountedState.ts) is a wrapper around `useEffect`. I looked for usages of these promise creators and I couldn't find uses that weren't wrapped in `useEffect` already.

I worry that this might introduce a "gotcha" if consumers of this API need to call the promises _before_ their component is mounted (i.e. in `useLayoutEffect`). If I understand this correctly, calling the promise will do nothing.

Reply:
`useMountedState` returns a predicate that indicates whether the component is mounted. This becomes important when the component unmounts while the promise is pending. Then, when the promise is resolved, the subsequent `setState` will cause an exception. Therefore all asynchronously called `setState`s should be guarded by such a check. I'm still trying to catch up with all the occurrences of that mistake in this plugin, so this is a small step.

Reply:
> I worry that this might introduce a "gotcha" if consumers of this API need to call the promises before their component is mounted (i.e. in useLayoutEffect). If I understand this correctly, calling the promise will do nothing.

As described above, the promise itself is not the problem. It resolving and thereby causing a `setState` call before mounting or after the unmounting is.